python/plugins/processing/tools/vector.py

```diff
@@ -60,6 +60,12 @@
     QgsWkbTypes.MultiPoint: 'MultiPoint',
     QgsWkbTypes.MultiLineString: 'MultiLineString',
     QgsWkbTypes.MultiPolygon: 'MultiPolygon',
+    QgsWkbTypes.Point25D: 'Point25D',
+    QgsWkbTypes.LineString25D: 'LineString25D',
+    QgsWkbTypes.Polygon25D: 'Polygon25D',
+    QgsWkbTypes.MultiPoint25D: 'MultiPoint25D',
+    QgsWkbTypes.MultiLineString25D: 'MultiLineString25D',
+    QgsWkbTypes.MultiPolygon25D: 'MultiPolygon25D',
 }
```

Comment:
Everything else looks good - but I wonder about using GEOM_TYPE_MAP in the creation of memory layers (eg line 570). Doesn't this break support for M/Z/ZM geometry types? That line (570) could be replaced again by using QgsWkbTypes.displayString() and then it's both future proof and also would handle all geometry types.